### DISCUSSION TITLE
Align 0 on X/Y axes

### DISCUSSION THREAD

**Discussion by PointSingularity** [recharts/recharts#3282] [Q&A] [answered]
Is there a way to align the axes so the 0 match?

```tsx
<YAxis
  dataKey="Height"
  type="number"
  tickCount={y.tickCount}
  interval={0}
  ticks={y.ticks}
  domain={['auto', 'auto']}

  allowDataOverflow={true}
/>;
<XAxis
  dataKey="Perimeter"
  domain={['auto', 'auto']}
  tickCount={x.tickCount}
  ticks={x.ticks}
  interval={0}
  type="number"

  allowDataOverflow={true}
/>;

```
![image](https://user-images.githubusercontent.com/56246797/214882543-884b97d9-3519-46d1-b3b5-199d0a11599f.png)

If I set the domain to [0, 'auto'] I get this, which is almost right, but the label is cut-off, and the lines are not shown below 0.
![image](https://user-images.githubusercontent.com/56246797/214887982-aef0855f-6f27-425b-b273-bb249a1d9efe.png)

Any insight would be greatly appreciated, thank you!

**Comment by ckifer** [comment, marked as answer]
Hi @PointSingularity - this is probably more of a feature request than anything. Recharts currently isn't great at adjusting the Y-axis below zero while still keeping components visible.

My only suggestion off the top of my head would be to add some margin bottom to the chart area or increase the chart height. 

Something that needs worked on for sure

**Reply by PointSingularity** [reply]
Gotcha, thanks for the answer!

It would be nice to have, but it's not a dealbreaker!

Just as an additional question, is there a way to force show the rest of the data points on that 2nd image?

**Reply by ckifer** [reply]
As I'm not one of the original creators I can honestly say I'm not sure. From what I've found there is usually a way to accomplish what you want though - if you create a simple code sandbox I'd be willing to try and help but I'll probably end up just doing trial and error 😅

**Reply by PointSingularity** [reply]
> As I'm not one of the original creators I can honestly say I'm not sure. From what I've found there is usually a way to accomplish what you want though - if you create a simple code sandbox I'd be willing to try and help but I'll probably end up just doing trial and error 😅

Thank you for your help, I really appreciate it 😄 

[Here is the code, so check it out if you have time.](https://stackblitz.com/edit/react-ts-dq9xqh?file=Chart.tsx,data.tsx,App.tsx)

**Reply by ckifer** [reply]
Yeah from an initial investigation this needs to be implemented internally. Can't really find any opportunities to customize and I think the X axis is what stops the overflow from continuing past 0. Feel free to make an issue with a feature request!

btw you don't need to calculate dataMin and dataMax if you don't want, you can do
```jsx
 domain={[0, 'dataMax']} - `'dataMin'` and `'dataMax'` are valid values
```

**Reply by PointSingularity** [reply]
Good to know!

Yeah, I'll make a feature request.

Thanks again 😃
